tests/test_evpn_l3_vxlan.py

```diff
@@ -0,0 +1,1726 @@
+        tbl = swsscommon.Table(conf_db, "VRF")
+        fvs = swsscommon.FieldValuePairs(attrs)
+        tbl.set(vrf_name, fvs)
+        time.sleep(2)
+
+        tbl = swsscommon.Table(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_VIRTUAL_ROUTER")
+        current_entries = set(tbl.getKeys())
+        assert len(current_entries - initial_entries) == 1
+
+        new_vr_ids  = get_created_entries(asic_db, self.ASIC_VRF_TABLE, self.vnet_vr_ids, 1)
+        self.vnet_vr_ids.update(new_vr_ids)
+        self.vr_map[vrf_name] = { 'ing':new_vr_ids[0], 'egr':new_vr_ids[0]}
+
+        return list(current_entries - initial_entries)[0]
+
+    def remove_vrf(self, dvs, vrf_name):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        tbl = swsscommon.Table(conf_db, "VRF")
+        tbl._del(vrf_name)
+        time.sleep(2)
+
+
+    def is_vrf_attributes_correct(self, db, table, key, expected_attributes):
+        tbl =  swsscommon.Table(db, table)
+        keys = set(tbl.getKeys())
+        assert key in keys, "The created key wasn't found"
+
+        status, fvs = tbl.get(key)
+        assert status, "Got an error when get a key"
+
+        # filter the fake 'NULL' attribute out
+        fvs = filter(lambda x : x != ('NULL', 'NULL'), fvs)
+
+        attr_keys = {entry[0] for entry in fvs}
+        assert attr_keys == set(expected_attributes.keys())
+
+        for name, value in fvs:
+            assert expected_attributes[name] == value, "Wrong value %s for the attribute %s = %s" % \
+                                  (value, name, expected_attributes[name])
+
+
+class TestL3Vxlan(object):
+
+    def get_vxlan_obj(self):
+        return VxlanTunnel()
+
+    def setup_db(self, dvs):
+        self.pdb = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+        self.adb = swsscommon.DBConnector(1, dvs.redis_sock, 0)
+        self.cdb = swsscommon.DBConnector(4, dvs.redis_sock, 0)
+
+#    Test 1 - Create and Delete SIP Tunnel and VRF VNI Map entries
+#    @pytest.mark.skip(reason="Starting Route Orch, VRF Orch to be merged")
+    @pytest.mark.dev_sanity
+    def test_sip_tunnel_vrf_vni_map(self, dvs, testlog):
+        vxlan_obj = self.get_vxlan_obj()
+
+        self.setup_db(dvs)
+        tunnel_name = 'tunnel_2'
+        map_name = 'map_1000_100'
+        vrf_map_name = 'evpn_map_1000_Vrf-RED'
+
+        vxlan_obj.fetch_exist_entries(dvs)
+
+        print "\n\nTesting Create and Delete SIP Tunnel and VRF VNI Map entries"
+        print "\tCreate SIP Tunnel"
+        create_vlan1(dvs,"Vlan100")
+        create_vxlan_tunnel(dvs, tunnel_name, '6.6.6.6')
+        create_evpn_nvo(dvs, 'nvo1', tunnel_name)
+        print "\tPass"
+
+        print "\tCreate Vlan-VNI map and VRF-VNI map"
+        create_vxlan_tunnel_map(dvs, tunnel_name, map_name, '1000', 'Vlan100')
+
+        vxlan_obj.create_vrf(dvs, "Vrf-RED")
+        create_vxlan_vrf_tunnel_map(dvs, 'Vrf-RED', '1000')
+        print "\tPass"
+
+        #vlanlist = ['100', '101', '102']
```

Comment:
If not required please remove the unwanted code.

Reply:
ok. Will do.